Documentation fixes & improvements

GiudGiud opened the pull request:
Please feel free to add to it

Misc fixes
More missing syntax pages, we're almost there.

GiudGiud committed:
Add more debug docs

GiudGiud committed:
Fix uniform refinement description in cli args list

cticenhour commented:
Ha - didn't see this before I pushed mine. Current changes don't seem to overlap, so I'll just leave them separate.

GiudGiud commented:
i have some unrelated commits. I might move my work over when I m done then

GiudGiud committed:
Move ExternalPetscProblem to same folder as the other problems

GiudGiud committed:
Add syntax documentation to the Periodic BCs, refs #15968

GiudGiud committed:
Add syntax documentation to the Nodal Kernels, refs #15968

GiudGiud committed:
Add basic documentation to aux scalar kernels refs #15968
Leans on auxkernels a lot. Any one using scalar will know fields already

GiudGiud committed:
Add syntax docs to deprecated action, refs #15968

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1572777/) on 8845fa8 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24546/site)

This comment will be updated on new commits.

GiudGiud committed:
Bounds syntax doc

GiudGiud committed:
Remove Robert's notes
The end of an era

GiudGiud committed:
Dirac Kernels syntax docs, refs  #15968

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1581813/) on 0d9f5fe wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>8dc8a5</th>
      <th colspan=4>#24546 0d9f5f</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.41%</td>
      <td>85.41%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>88799</td>
      <td>88803</td>
      <td>+4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15169</td>
      <td>15165</td>
      <td>-4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24546/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24546/coverage/framework)


## Modules coverage

### External petsc solver

<table>
  <thead>
    <tr>
      <th></th>
      <th>8dc8a5</th>
      <th colspan=4>#24546 0d9f5f</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.45%</td>
      <td>85.45%</td>
      <td>-</td>
      <td>85.51%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>370</td>
      <td>370</td>
      <td>-</td>
      <td>59</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>63</td>
      <td>63</td>
      <td>-</td>
      <td>10</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24546/coverage_diff/external_petsc_solver)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24546/coverage/external_petsc_solver)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24546/coverage/xfem)

</details>

## Warnings

- `external_petsc_solver` new line coverage rate 85.51% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud committed:
Add quadrature syntax doc, refs #15968

GiudGiud committed:
Add nodal normal syntax doc, refs #15968

GiudGiud committed:
Symlink the adaptivity syntax page for the Executioner/adaptivity page

GiudGiud committed:
Add syntax docs to boundary conditions, refs #15968

GiudGiud committed:
Finish source documentation for the framework, refs #15968

lindsayad commented on framework/doc/content/source/actions/AddPeriodicBCAction.md:
```diff
@@ -1,29 +1,19 @@
 
 !listing auto_periodic_bc_test.i block=BCs
 
-## Translation Periodic Boundaries
-
-!listing periodic/periodic_bc_test.i block=BCs
-
-## Function Transform Periodic Boundaries
-
-!listing periodic/trapezoid.i block=Functions BCs
```
Why remove this? Even without any descriptive text I think this could be useful

lindsayad replied:
Ah never mind, I see it moved

lindsayad commented on framework/doc/content/source/bcs/DirichletBC.md:
```diff
@@ -33,6 +33,12 @@ before the solve begins. With `preset = false`, the boundary condition is
 
+## Similar Dirichlet BCs
+
+To use a Function instead of a constant value : [FunctionDirichletBC.md]
+To impose a Dirichlet BC using a penalty method or on an elemental variable : [PenaltyDirichletBC.md]
+To use a another variable instead of a constant value : [MatchedValueBC.md]
```
```suggestion
- To use a Function instead of a constant value : [FunctionDirichletBC.md]
- To impose a Dirichlet BC using a penalty method or on an elemental variable : [PenaltyDirichletBC.md]
- To use a another variable instead of a constant value : [MatchedValueBC.md]
```

lindsayad commented on framework/doc/content/source/outputs/OutputWarehouse.md:
```diff
@@ -1,3 +1,19 @@
 # OutputWarehouse
 
-!alert construction title=Undocumented
+The `OutputWarehouse` is a storage container for all [Outputs](syntax/Outputs/index.md)
```
```suggestion
The `OutputWarehouse` is a storage container for all [Output](syntax/Outputs/index.md)
```

lindsayad commented on framework/doc/content/source/outputs/OutputWarehouse.md:
```diff
@@ -1,3 +1,19 @@
+
+In addition to keeping track of all `Outputs` objects, the `OutputWarehouse` also:
+
+- keep track of the `sync_times` at which output is forced
```
```suggestion
- keeps track of the `sync_times` at which output is forced
```

lindsayad commented on framework/doc/content/source/outputs/OutputWarehouse.md:
```diff
@@ -1,3 +1,19 @@
+In addition to keeping track of all `Outputs` objects, the `OutputWarehouse` also:
+
+- keep track of the `sync_times` at which output is forced
+- handle the output to the console. The `ConsoleStream` that many objects use to output, uses the `OutputWarehouse`
```
```suggestion
- handles the output to the console. The `ConsoleStream` that many objects use to output, uses the `OutputWarehouse`
```

lindsayad commented on framework/doc/content/source/variables/MooseVariableFE.md:
```diff
@@ -2,7 +2,16 @@
 !syntax description /Variables/MooseVariable
 
-!alert construction title=Undocumented
+A `MooseVariableFE` is a derived class of a `MooseVariable`. It is a finite element field variable.
```
```suggestion
A `MooseVariableFE` is a derived class of a `MooseVariableField`. It is a finite element field variable.
```

lindsayad commented on framework/doc/content/syntax/BCs/index.md:
```diff
@@ -1,29 +1,49 @@
 
-!syntax list /BCs objects=True actions=False subsystems=False
+We show below code snippets for the `DirichletBCBase`. This boundary condition sets the value of a variable on
+a node. The boundary conditions are im
```
I'm waiting in suspense!

GiudGiud replied:
not sure what I wanted to write. if it comes back to me I ll add it to another PR

lindsayad commented on framework/doc/content/syntax/Bounds/index.md:
```diff
@@ -1,29 +1,33 @@
 
-## Example Input File Syntax
+Note that in order for `Bounds` to have an effect, the user has to specify the
+PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither options are specified. The PETSc manual pages for the `vinewtonssls` algorithm
```
```suggestion
PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither option is specified. The PETSc manual pages for the `vinewtonssls` algorithm
```

lindsayad commented on framework/doc/content/syntax/Debug/index.md:
```diff
@@ -58,6 +59,23 @@ interact in arbitrarily complex ways on a group of variables are indeed executed
+## Viewing objects created by an applications
+
+Numerous applications will use [Actions](actions/Action.md) to simplify the user input. This reduces opportunities for
+mistakes in an input, but has the inconvenient of hiding part of the simulation setup. The [!param](/Debug/show_actions) will,
```
```suggestion
mistakes in an input, but has the inconvenience of hiding part of the simulation setup. The [!param](/Debug/show_actions) will,
```

lindsayad commented on framework/doc/content/syntax/Debug/index.md:
```diff
@@ -58,6 +59,23 @@ interact in arbitrarily complex ways on a group of variables are indeed executed
+
+
+Additionally, [!param](/Debug/show_execution_order) will provide the list of objects executed as they are executed.
+This includes, [Kernels](syntax/Kernels/index.md) (and declinations), [AuxKernels](syntax/AuxKernels/index.md), [boundary conditions](syntax/BCs/index.md)
```
declinations? I had to search the definition of this word and it doesn't seem like a match for this context

GiudGiud replied:
changed

roystgnr commented on framework/doc/content/syntax/BCs/Periodic/index.md:
```diff
@@ -1,29 +1,47 @@
+- Can be restricted to specific variables.
 
-!! Replace this line with information regarding the Periodic system.
+Supports arbitrary translation vectors for defining periodicity.
```
MOOSE supports more than just the translation-by-fixed-vector case, right?  `FunctionPeriodicBoundary` will give you translations by arbitrary parsed functions.

GiudGiud replied:
ah that s what I meant to convey. i ll reword

roystgnr commented on framework/doc/content/syntax/BCs/index.md:
```diff
@@ -1,29 +1,49 @@
-## Overview
+Some of the main types of boundary conditions are:
+- [Dirichlet Boundary conditions](DirichletBC.md) to set the value of a variable on a boundary
+- [Neumann Boundary conditions](NeumannBC.md) to set a condition on the derivative of a variable. Depending on the
```
Strictly speaking a Neumann BC is a condition on the flux of a variable; that's only equal to the derivative for simple problems like the Laplacian.  And looking at `NeumannBC.C`, it looks to me like it's applying a forcing that would correspond to the flux, not the derivative, in cases where a constitutive term makes those differ.

GiudGiud replied:
I was thinking that too but wikipedia disagrees
https://en.wikipedia.org/wiki/Neumann_boundary_condition

GiudGiud replied:
I ll reword to make that clear

roystgnr replied:
> always

Now I'm kind of wondering about exceptions; the theorems I recall always started with something like "let c>0 = min_Ω (k)", but I'm not sure whether disallowing any singularity in k was a *necessary* postulate or just a *simple* one...

GiudGiud replied:
I think the controversy for me is more around fluxes that are not derivatives, even with a (reasonable) coefficient.
The wording right now should be safe?

roystgnr replied:
Looks reasonable now.

lindsayad commented on modules/external_petsc_solver/doc/content/source/problems/problem/ExternalPETScProblem.md:
`problems/problem` doesn't seem right

GiudGiud replied:
I ll fix it

roystgnr commented on framework/doc/content/syntax/Bounds/index.md:
```diff
@@ -1,29 +1,33 @@
-point for documenting the system, which includes the typical automatic documentation associated with
-a system; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
+The `Bounds` system is design to bound the value of a nonlinear variable. Whether the bound is an upper or lower
```
```suggestion
The `Bounds` system is designed to bound the value of a nonlinear variable. Whether the bound is an upper or lower
```

roystgnr reviewed (changes requested):
I started out making comments rather than an official review, because I couldn't find anything that didn't look nitpicky, but I think the Quadrature doc fixes are important.

GiudGiud commented on framework/doc/content/syntax/Executioner/Quadrature/index.md:
```diff
@@ -1,29 +1,47 @@
-# Quadrature System
+The `Quadrature` order is chosen by default so that the product of the test function and the shape functions is integrated exactly.
+When using Galerkin's method, this means the quadrature would need to integrate exactly polynomials of order
+twice the order of the variables.
```
Interesting. I guess it s free with the Gauss quadrature.

GiudGiud replied:
changed

GiudGiud commented on framework/doc/content/syntax/Executioner/Quadrature/index.md:
```diff
@@ -1,29 +1,47 @@
+In 1D, the following quadratures are known to integrate exactly:
 
-!! Replace this line with information regarding the Quadrature system.
+- GAUSS (-Legendre), the default rule, of order n integrates exactly 1D polynomials of order up to 2n-1
+- GAUSS_LOBATTO of order n integrates exactly 1D polynomials of order up to 2n-3
```
wikipedia agress with me
https://en.wikipedia.org/wiki/Gaussian_quadrature#Gauss%E2%80%93Kronrod_rules

GiudGiud replied:
ah no it doesnt
https://en.wikipedia.org/wiki/Gauss%E2%80%93Kronrod_quadrature_formula

GiudGiud commented on framework/doc/content/syntax/Bounds/index.md:
```diff
@@ -1,29 +1,33 @@
 
-## Overview
+!alert note
+Only constant elemental variables are supported at this time.
```
yeah my bad

GiudGiud commented on framework/doc/content/syntax/Bounds/index.md:
```diff
@@ -1,29 +1,33 @@
 
-## Overview
+!alert note
+Only constant elemental variables are supported at this time.
```
```suggestion
Only nodal and constant elemental variables are supported at this time.
```

GiudGiud commented on framework/doc/content/syntax/Debug/index.md:
```diff
@@ -58,6 +59,23 @@ interact in arbitrarily complex ways on a group of variables are indeed executed
+
+
+Additionally, [!param](/Debug/show_execution_order) will provide the list of objects executed as they are executed.
+This includes, [Kernels](syntax/Kernels/index.md) (and declinations), [AuxKernels](syntax/AuxKernels/index.md), [boundary conditions](syntax/BCs/index.md)
```
```suggestion
This includes, [Kernels](syntax/Kernels/index.md) (and derived classes), [AuxKernels](syntax/AuxKernels/index.md), [boundary conditions](syntax/BCs/index.md)
```

GiudGiud committed:
Address Alex and Roy's reviews

- a quadrature of n points does not mean order n
- better grammar, less typos, more accurate language

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>
Co-authored-by: roystgnr <roy@stogners.org>

Update framework/doc/content/syntax/NodalKernels/index.md

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

lindsayad commented on framework/doc/content/syntax/Debug/index.md:
```diff
@@ -58,6 +59,23 @@ interact in arbitrarily complex ways on a group of variables are indeed executed
+
+
+Additionally, [!param](/Debug/show_execution_order) will provide the list of objects executed as they are executed.
+This includes, [Kernels](syntax/Kernels/index.md) (and derived classes), [AuxKernels](syntax/AuxKernels/index.md), [boundary conditions](syntax/BCs/index.md)
+(and declinations), [UserObjects](syntax/UserObjects/index.md), [Postprocessors](syntax/Postprocessors/index.md) and
```
I think that derived classes is kind of implicit. If it's not, then it seems like it should be consistently applied to all the systems listed here. (There is also one more "declinations")

GiudGiud replied:
ok I ll reword
they are not derived classes actually, what was I thinking

lindsayad reviewed (approved):
thank you for these additions/fixes!

GiudGiud commented:
Failures are unrelated (pulling saline and a timeout)

GiudGiud commented:
Thanks for the reviews!
